File: Python/Populating Next Right Pointers in Each Node II.py
def connect(self, root):
    dummy = TreeNode(0)
    head = root
    while head:
        cur = dummy
        dummy.next = None
        while head:
            if head.left:
                cur.next = head.left
                cur = cur.next
            if head.right:
                cur.next = head.right
                cur = cur.next
            head = head.next
        head = dummy.next 

File: Java/src/Populating Next Right Pointers in Each Node II.java
public class Solution {
    public void connect(TreeLinkNode root) {
        while(root != null){
            TreeLinkNode dummy = new TreeLinkNode(0);
        	TreeLinkNode cur = dummy;
            while(root!=null){
                if(root.left != null) {
                    cur.next = root.left; cur = cur.next;
                }
                if(root.right != null) {
                    cur.next = root.right; cur = cur.next;
                }
                root = root.next;
            }
            root = dummy.next;
        }
    }
}